asab/web/container.py

```diff
@@ -136,6 +137,19 @@ def __init__(self, websvc: WebService, config_section_name: str, config: typing.
 			preflight_paths = re.split(r"[,\s]+", preflight_str, re.MULTILINE)
 			self.add_preflight_handlers(preflight_paths)
 
+		@aiohttp.web.middleware
+		async def set_request_context(request: aiohttp.web.Request, handler):
+			"""
+			Make sure that the incoming aiohttp.web.Request is available via Request context variable
+			"""
+			request_ctx = Request.set(request)
+			try:
+				return await handler(request)
+			finally:
+				Request.reset(request_ctx)
+
+		self.WebApp.middlewares.append(set_request_context)
```

Comment:
i'm not sure if `asab.web.container` is the right place for this piece of code

Reply:
Why is that needed?

Reply:
at the moment mostly because of `DiscoveryService.session()`. most discovery session calls are (or soon will need to be) authorized which means you have to pass the authorization from the incoming request to the outgoing request. instead of requiring devs to always pass on the `request` object (like `DiscoveryService.session(auth=request)`), it can be taken automatically from the context variable. also, HTTP request objects should not be passed from the handler layer to the service layer.

Reply:
I am already using this :)